astropy/cosmology/core.py

```diff
@@ -2857,12 +2870,12 @@ def _float_or_none(x, digits=3):
     return fmtstr.format(x)
 
 
-def vectorize_if_needed(func, x):
+def vectorize_if_needed(func, *x):
     """ Helper function to vectorize functions on array inputs"""
-    if isiterable(x):
-        return np.vectorize(func)(x)
+    if any(map(isiterable, x)):
+        return np.vectorize(func)(*x)
```

Comment:
Thanks! Already in the morning I can answer the question "what have I learned today?".